build/vega-lite-schema.json

```diff
@@ -2,13 +2,79 @@
     "$schema": "http://json-schema.org/draft-04/schema#",
     "anyOf": [
         {
-            "$ref": "#/definitions/GenericUnitSpec<string | MarkDef, EncodingWithFacet>"
+            "allOf": [
+                {
+                    "$ref": "#/definitions/GenericUnitSpec<string | MarkDef, EncodingWithFacet>"
+                },
+                {
+                    "$ref": "#/definitions/TopLevelProperties"
+                },
+                {
+                    "additionalProperties": false,
```

Comment:
(updated comment)

@domoritz 

My declaration [in the code here](https://github.com/vega/vega-lite/pull/2043/files#diff-f0bc433010488ea7f9df55b35517b91bR20) seems reasonable. 

However, the output JSON schema actually cause the validation to stupidly fail because each part of the `allOf` has `additionalProperties: false`.  (See error https://github.com/vega/vega-lite/pull/2043/files#diff-f0bc433010488ea7f9df55b35517b91bR20.) 

There are two solutions here:
a) Fix it in the JSON schema generator.  Basically, just like interface `extends` we could just generate a merged object instead of using `allOf`.

b) Don't do `TopLevel<S extends BaseSpec>`.  Instead, for every occurrence of `TopLevel<XXX>`, declare a custom TopLevelXXX that `extends XXX, TopLevelProperties`.

I think a) is a cleaner solution if it's not too hard to do.  (Currently, `&` will always break the output schema if `additionalProperties` is globally set `false` anyway. 
On the other hand, b) is will lead to many tedious and repetitive declarations. 
I think a) is a cleaner solution if it's not too hard to do.  b) is obviously a hack and will lead to  repetitive declarations (That will add 5-10 more repeated declarations in `spec.ts`). 


I could do b) but I want to check with you if you want to look into a) first.  Let me know!